src/utils/SpectraSTSearchAdapter.cpp

```diff
@@ -0,0 +1,373 @@
+  public TOPPBase
+{
+ public:
+  // Define parameter name
+  static const String param_executable;
+
+  static const String param_spectra_files;
+  static const String param_spectra_files_formats;
+  static const String param_library_file;
+  static const String param_sequence_database_file;
+  static const String param_sequence_database_type;
+  static const String param_search_file;
+  static const String param_params_file;
+  static const String param_precursor_mz_tolerance;
+  static const String param_use_isotopically_averaged_mass;
+  static const String param_use_all_charge_states;
+  static const String param_output_files;
+  static const vector<String> param_output_file_formats;
+  static const vector<String> param_input_file_formats;
+  static const String param_user_mod_file;
+
+
+  TOPPSpectraSTSearchAdapter() :
+    TOPPBase("SpectraSTSearchAdapter", "Interface to the SEARCH Mode of the SpectraST executable", false)
+  {
+
+  }
+
+
+protected:
+
+  // this function will be used to register the tool parameters
+  // it gets automatically called on tool execution
+  void registerOptionsAndFlags_()
+  {
+      StringList empty;
+
+      // Handle executable
+      registerInputFile_(TOPPSpectraSTSearchAdapter::param_executable, "<path>", "spectrast", "Path to the SpectraST executable to use; may be empty if the executable is globally available.", true, false, ListUtils::create<String>("skipexists"));
+
+      // register spectra input files
+      registerInputFileList_(TOPPSpectraSTSearchAdapter::param_spectra_files, "<SearchFileName1> [ <SearchFileName2> ... <SearchFileNameN> ]", empty, "Name(s) of file containing unknown spectra to be searched.", true, false);
+      setValidFormats_(TOPPSpectraSTSearchAdapter::param_spectra_files, ListUtils::create<String>(TOPPSpectraSTSearchAdapter::param_spectra_files_formats), false);
+
+      // register Output files
+      registerOutputFileList_(TOPPSpectraSTSearchAdapter::param_output_files,"<OutputFile1> [ <OutputFileName2> ... <OutputFileNameN> ]", empty, "Files where the output should be written to. Make sure to specify one output file for each input file.",true,false);
+      setValidFormats_(TOPPSpectraSTSearchAdapter::param_output_files, TOPPSpectraSTSearchAdapter::param_output_file_formats, false);
+
+
+      // Require library file to be searched  // TODO Check the existence of the .spidx file
+      registerInputFile_(TOPPSpectraSTSearchAdapter::param_library_file, "<lib_file>.splib", "", "Specify library file.", true, false);
+      setValidFormats_(TOPPSpectraSTSearchAdapter::param_library_file,ListUtils::create<String>("splib"),false);
+
+
+      // Sequence database file
+      registerInputFile_(TOPPSpectraSTSearchAdapter::param_sequence_database_file, "<sequencedb_file>.fasta", "", "Specify a sequence database file",false,false);
+      setValidFormats_(TOPPSpectraSTSearchAdapter::param_sequence_database_file, ListUtils::create<String>("fasta"),false);
```

Comment:
hint: no search and replace because e.g. "," in StringList may not be separated by space